src/server.jsx

```diff
@@ -16,22 +16,35 @@ const bundles = [
   '/index.bundle.js'
 ];
 
+const Content = require('./_content.json');
+
+const paths = Content.children.reduce((paths, page) => {
+  if (page.type === 'directory') {
+    page.children.forEach(child => (paths[child.url] = child.title));
+  } else {
+    paths[page.url] = page.title;
+  }
+
+  return paths;
+}, {});
```

Comment:
Does it makes sense to generate a `_paths.json` file after `_content.json` is created? We could import it and avoid doing this every time the server file runs, it might improve performance on SSR considering we have lots of pages.

Reply:
If this is the case Fernando, we can leave it for further optimization and I can tackle that once this is merged :)

Reply:
I think better than another file would just be to have a `walk` or `find` utility to find the page node within the main tree (rather than creating a separate array and walking it).

Reply:
Good, to make it clear this is not a blocker for the PR to be merged.

Reply:
Yeah, I think once we've tackled everything I may try to spend a day or two just doing some cleanup and extraction of code to other packages so that it's fairly clean before going into `master`.

Reply:
In #2108, I added a `content-utils.js` file that should make things like this a breeze using the `WalkContent` or `FindInContent` method. I think we should get #2108 and #2074 merged into `rebuild` and then really start a push to finish up the rest of these. From what I've seen, most of the `rebuild` PRs are pretty close and may just need a little refactoring and abstracting to a separate package (e.g. `remark` plugins). Skimming over them again now.